app/pull-gitlab-variables.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# arguments:
+# - ATLANTIS_GITLAB_TOKEN
+# - HEAD_REPO_OWNER
+# - HEAD_REPO_NAME
+# - ALLOWLIST_FILE e.g. GOOGLE_APPLICATION_CREDENTIALS
+# - ALLOWLIST_ENV_VAR e.g. AWS_ACCESS_KEY_ID,AWS_SECRET_ACCESS_KEY
+# - REPO_REL_DIR e.g environments/dev/cicd/azure
+
+# Variable needed for `glab`
+export GITLAB_TOKEN=${ATLANTIS_GITLAB_TOKEN}
+
+ENV_SCOPE=$(echo $REPO_REL_DIR | sed -nE 's/.*\/(dev|test|prod|staging)\/.*/\1/p')
+: "${ENV_SCOPE:=*}"
+
+ENCODED_REPO_NAME=$(jq -rn --arg x "${HEAD_REPO_OWNER}/${HEAD_REPO_NAME}" '$x|@uri')
+API_RESPONSE=$(glab api projects/$ENCODED_REPO_NAME/variables)
+MULTIENV_RESULT=""
+
+
+ALLOWLIST_ENV_VAR_ARRAY=($(printf '%s\n' ${ALLOWLIST_ENV_VAR//,/ } | sort))
```

Comment:
Shouldn't we have some default values perhaps?

Reply:
yeah, I'll add something there ☺️

Reply:
(added)